### PR TITLE
Add arbitrary instances for newtypes in Prelude

### PR DESCRIPTION
**Pull request by JordanMartinez**
I didn't include Endo because it would take more time for me to figure out what it should be. The others are pretty obvious, so I just did those.

### CODE REVIEW

**Comment by kl0tl** [comment]
I think we can adapt the Arbitrary instance for `a -> b` to `Endo a`:

```purs
instance arbEndo :: (Arbitrary a, Coarbitrary a) => Arbitrary (Endo a) where
  arbitrary = Endo <$> repeatable (\a -> coarbitrary a arbitrary)
```

**Comment by hdgarrood** [comment]
Is there a reason you didn’t do `arbitrary = Endo <$> arbitrary`?

**Comment by JordanMartinez** [comment]
Thanks all! I've added the `Endo` instance.

**Comment by JordanMartinez** [comment]
#120 needs to be merged before this will pass.

File: src/Test/QuickCheck/Arbitrary.purs [hunk 1]
```diff
@@ -143,6 +151,30 @@ instance coarbFunction :: (Arbitrary a, Coarbitrary b) => Coarbitrary (a -> b) w
...
+instance arbConj :: Arbitrary a => Arbitrary (Conj a) where
+  arbitrary = Conj <$> arbitrary
+
+instance arbDisj :: Arbitrary a => Arbitrary (Disj a) where
+  arbitrary = Disj <$> arbitrary
+
+instance arbDual :: Arbitrary a => Arbitrary (Dual a) where
+  arbitrary = Dual <$> arbitrary
+
+instance arbEndo :: (Arbitrary (c a a), Arbitrary a) => Arbitrary (Endo c a) where
```
Comment (kl0tl) [inline src/Test/QuickCheck/Arbitrary.purs:175]: The `Arbitrary a` constraint seems redundant given `Arbitrary (c a a)`.

**Comment by kl0tl** [comment]
> Is there a reason you didn’t do `arbitrary = Endo <$> arbitrary`?

I blame the lack of caffeine!

**Comment by JordanMartinez** [comment]
CI now builds.

**Comment by JordanMartinez** [comment]
Can this get merged?

**Review by thomashoneyman — APPROVED** [review]

**Pull request merged**